dnadoh/recombine.py

```diff
@@ -0,0 +1,56 @@
+"""Recombination of the genomes of two people
+to produce a third person. Other features are
+averaged with some noise.
+"""
+
+from typing import List
+from .synthesize import PersonGenerator, Person
+
+def recombine(
+    person1: Person,
+    person2: Person,
+    reference_genome: str,
+    pid: str,
+    recombination_prob: List[float],
+    recombination_loci: List[int]
+    ) -> Person:
+    """Recombine two genomes"""
+    # Error checking
+    assert len(recombination_loci) == len(recombination_prob)
+    assert len(person1.genome) == len(person2.genome)
+    assert all(recombination_loci >= 0 & recombination_loci < len(person1.genome))
+    assert all(recombination_prob <= 1 & recombination_prob >= 0)
+
+    # Initializing
+    genome1 = person1.genome
+    genome2 = person2.genome
+
+    # If the index 0 is not in the list of recombination loci
+    # Add it to initialize recombination by selecting either genome
+    # with equal probability
+    if 0 not in recombination_loci:
+        recombination_loci = [0] + recombination_loci
+        recombination_prob = [0.5] + recombination_prob
+    # Adding the last coordinate of the dna to provide an end index
+    # for the last segment that can be recombined
+    if len(genome1) not in recombination_loci:
+        recombination_loci = recombination_loci + [len(genome1)]
+        recombination_prob = recombination_prob + [0]
+
+    ## Ordering
```

Comment:
are these two lines making sure that the loci are sorted in ascending order, and that the probabilities are sorted accordingly so that they still match up with their intended loci?

Reply:
Yes, I think I might steal some of these interpretations as code comments 😄